apps/prairielearn/src/models/enrollment.ts

```diff
@@ -123,26 +185,29 @@ export async function ensureCheckedEnrollment({
   await ensureEnrollment({
     course_instance_id: course_instance.id,
     user_id: authz_data.authn_user.user_id,
+    agent_user_id: authz_data.authn_user.user_id,
+    agent_authn_user_id: authz_data.user.id,
+    action_detail,
   });
 }
 
-export async function getEnrollmentForUserInCourseInstance({
+export async function selectOptionalCourseInstanceEnrollmentByUserId({
```

Comment:
Nit: name this `selectOptionalEnrollmentByUserId` for consistency with `selectOptionalEnrollmentByUid`. The "course instance" part is implicit, enrollments always exist in the context of a course instance.